### DISCUSSION TITLE
Why would I ever use a separation mapping?

### DISCUSSION THREAD

**Discussion by nkiesel** [komapper/komapper#508] [Q&A] [answered]
Looks to me that using self-mapping requires less code and better readability.  So why would I ever use separation mapping? Only reason I could see is if I cannot modify the entity class definitions.  Or are there other reasons?

**Comment by nakamura-to** [comment, marked as answer]
Separation mapping is useful when we want to use same entity classes in multiple projects, e.g., front-end and back-end projects. With separation mapping, the front-end project does not need to depend on Komapper's annotation module.

**Reply by nkiesel** [reply]
Thx, makes sense. I would still think that the quickstart documentation should use self-mapping, because that reduces the barrier of entry.

**Reply by nakamura-to** [reply]
Thanks for your feedback. It's a very good idea. I have changed the Quickstart documentation.
